src/smart-components/platform/platform-templates.js

```diff
@@ -1,133 +1,163 @@
+      ...meta,
+      offset: 0
+    });
+  };
 
-  handleOnPerPageSelect = limit => this.props.fetchPlatformItems(this.props.match.params.id, {
-    offset: this.props.paginationCurrent.offset,
+  const handleOnPerPageSelect = limit => fetchPlatformItems(id, {
+    offset: meta.offset,
     limit
   });
 
-  handleSetPage = (number, debounce) => {
+  const handleSetPage = (number, debounce) => {
     const options = {
-      offset: getNewPage(number, this.props.paginationCurrent.limit),
-      limit: this.props.paginationCurrent.limit
+      offset: getNewPage(number, meta.limit),
+      limit: props.paginationCurrent.limit
     };
-    const request = () => this.props.fetchPlatformItems(this.props.match.params.id, options);
+
+    const request = () => dispatch(fetchPlatformItems(id, filterValue, options));
     if (debounce) {
       return debouncePromise(request, 250)();
     }
 
     return request();
-  }
-
-  handleFilterChange = filterValue => this.setState({ filterValue });
+  };
 
-  render() {
+  const renderItems = () => {
+    const paginationCurrent = meta || defaultSettings;
     let filteredItems = {
-      items: this.props.platformItems
-      .filter(item => filterServiceOffering(item, this.state.filterValue))
-      .map(data => <PlatformItem key={ data.id } { ...data } />),
-      isLoading: this.props.isPlatformDataLoading
+      items: data ? data.map(item => <PlatformItem key={ item.id } { ...item } />) : [],
+      isLoading: isFetching || isFiltering
     };
 
-    let title = this.props.platform ? this.props.platform.name : '';
+    let title = platform ? platform.name : '';
     return (
       <Fragment>
         <ToolbarRenderer schema={ createPlatformsTopToolbarSchema({
           title,
           paddingBottom: false,
-          tabItems: this.tabItems
+          tabItems
         }) }/>
         <ToolbarRenderer schema={ createPlatformsFilterToolbarSchema({
-          onFilterChange: this.handleFilterChange,
-          searchValue: this.state.filterValue,
+          onFilterChange: handleFilterChange,
+          searchValue: filterValue,
           pagination: {
-            itemsPerPage: this.props.paginationCurrent.limit,
-            numberOfItems: this.props.paginationCurrent.count || 50,
-            onPerPageSelect: this.handleOnPerPageSelect,
-            page: getCurrentPage(this.props.paginationCurrent.limit, this.props.paginationCurrent.offset),
-            onSetPage: this.handleSetPage,
+            itemsPerPage: paginationCurrent.limit,
+            numberOfItems: paginationCurrent.count,
+            onPerPageSelect: handleOnPerPageSelect,
+            page: getCurrentPage(paginationCurrent.limit, paginationCurrent.offset),
+            onSetPage: handleSetPage,
             direction: 'down'
           }
         }) }/>
-        <ContentGallery { ...filteredItems }/>
+        <ContentGallery title={ title }
+          isLoading={ isFetching || isFiltering }
```

Comment:
This is a duplicate props. It is also present in the filtered items.

I think it should be removed from `filteredItems` and assign both props explicitly, not by using the spread operator.